source/MaterialXGenShader/Nodes/HwImageNode.cpp

```diff
@@ -0,0 +1,72 @@
+//
+// TM & (c) 2019 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXGenShader/Nodes/HwSourceCodeNode.h>
+#include <MaterialXGenShader/Nodes/HwImageNode.h>
+#include <MaterialXGenShader/HwShaderGenerator.h>
+#include <MaterialXGenShader/Util.h>
+
+#include <iostream>
+
+namespace MaterialX
+{
+// Additional implementaton arguments for image nodes
+const string UV_SCALE = "uv_scale";
+const string UV_OFFSET = "uv_offset";
+
+ShaderNodeImplPtr HwImageNode::create()
+{
+    return std::make_shared<HwImageNode>();
+}
+
+void HwImageNode::addInputs(ShaderNode& node, GenContext&) const
+{
+    // Add additional scale and offset inputs to match implementation arguments
+    ShaderInput* input = node.addInput(UV_SCALE, Type::VECTOR2);
+    input->setValue(Value::createValue<Vector2>(Vector2(1.0f, 1.0f)));
+    input = node.addInput(UV_OFFSET, Type::VECTOR2);
+    input->setValue(Value::createValue<Vector2>(Vector2(0.0f, 0.0f)));
+}
+
+void HwImageNode::setValues(const Node& node, ShaderNode& shaderNode, GenContext& context) const
```

Comment:
We added custom source code handling to allow insertion of a transform before lookup. This is currentlu used for UDIM transforms from the original space to the normalized space if we decide to use a texture atlas.

Reply:
That sounds like a good addition!